src/Query/BoolQuery.php

```diff
@@ -111,7 +111,13 @@ public function toArray()
             $mustContainer = $this->container[self::MUST];
             $query = array_shift($mustContainer);
 
-            return [$query->getType() => $query->toArray()];
+            if ($query->getType() == 'match') {
```

Comment:
This won't fix the bug. The match query is just like an example of any query. This bug occurs on all other queries also.